Add media playback badge for Area card

winterborn opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue or discussion
  in the additional information section.
-->

This PR implements issue #26684 by adding a media playback badge to Area cards, similar to existing motion/occupancy badges. The badge shows when media players in an area are playing or paused, providing quick visual feedback about active media in each room.

**Key Features:**
- Shows play/pause icon badge when media players are active in an area
- Displays appropriate label ("Media playing" or "Media paused")
- Only appears in compact mode, following existing badge patterns
- Respects area boundaries (only shows media players assigned to that specific area)

<img width="280" height="80" alt="image" src="https://github.com/user-attachments/assets/1471f32f-f044-4d28-a220-af4538ca4c21" />

<img width="280" height="80" alt="image" src="https://github.com/user-attachments/assets/f8c4e7ba-ffb4-45b5-ba68-04e64936502d" />


## Type of change
<!--
  What type of change does your PR introduce to the Home Assistant frontend?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR.
-->

```yaml
# No configuration needed - works automatically with existing Area cards
# Just ensure media players are assigned to areas in Settings > Areas & Zones
```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #26684
- This PR is related to issue or discussion: https://github.com/home-assistant/frontend/issues/26684
- Link to documentation pull request:

**Files Changed:**
- `src/common/entity/compute_active_media_states.ts` (new utility function)
- `src/panels/lovelace/cards/hui-area-card.ts` (badge integration)
- `test/common/entity/compute_active_media_states.test.ts` (test suite)

**Testing:**
- Added test cases covering happy path and edge cases
- Manually tested with media players in different states
- Verified badge appears/disappears based on media player state changes
- Confirmed area boundary respect (only shows for assigned media players)

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

<!--
  Thank you for contributing <3
-->

[docs-repository]: https://github.com/home-assistant/home-assistant.io

winterborn committed:
Add media playback badge for Area card

- Add computeActiveMediaStates utility function to find active media players in an area
- Add media badge rendering to Area card in compact mode
- Show play/pause icon with appropriate label based on media state
- Add comprehensive test suite covering happy path and edge cases
- Follow existing codebase patterns for alert badges and utility functions

Fixes #26684

bramkragten commented on src/common/entity/compute_active_media_states.ts:
```diff
@@ -0,0 +1,28 @@
+import type { HomeAssistant } from "../../types";
+import { generateEntityFilter } from "./entity_filter";
+
+export function computeActiveMediaStates(
```
Can we move this to `/data/media-player.ts` to keep all media player related logic in one place?

bramkragten commented on src/common/entity/compute_active_media_states.ts:
```diff
@@ -0,0 +1,28 @@
+): HassEntity[] {
+  const area = areaId ? hass.areas[areaId] : undefined;
+  if (!area) {
+    return [];
```
`areaId` is optional, and the function name doesn't imply that this function will only work for areas, and yet if you don't pass an `areaId` the function returns an empty array?

Make the name of the function more descriptive, like get `computeActiveAreaMediaStates`

bramkragten commented on src/panels/lovelace/cards/hui-area-card.ts:
```diff
@@ -302,6 +307,27 @@ export class HuiAreaCard extends LitElement implements LovelaceCard {
+      return nothing;
+    }
+
+    const firstState = states[0];
```
Why do we use the first state? Shouldn't `playing` take president over `paused` for example?

winterborn committed:
Address PR feedback: move function to data module and fix precedence logic

- Move computeActiveAreaMediaStates to src/data/media-player.ts
- Make areaId required parameter and rename function for clarity
- Fix precedence logic: playing state takes precedence over paused
- Update tests and remove redundant validation

silamon reviewed (commented):
It works like it should be, unfortunately we don't have a good way to display multiple tile badges.

winterborn committed:
Prioritize alert badges over media badges in area cards

winterborn commented:
> Yeah, those design processes are quite slow unfortunately. I think that we can solve easily without UX if you ensure that the alert tile badge is display with a higher priority than the music playing/paused.

Added [here](https://github.com/home-assistant/frontend/pull/26893/commits/72f1a1cc8ba24b62ba14f9644fbe094d063a2a31)

bramkragten commented:
I think we should not show paused media players, like Sonos media players will always be paused if they are not playing. We should only show it when it is playing media.

winterborn committed:
Simplify media badge to only show playing media

bramkragten commented on src/data/media-player.ts:
```diff
@@ -520,3 +522,33 @@ export const mediaPlayerJoin = (
+ * Compute active media player states in a specific area.
+ * @param hass Home Assistant object
+ * @param areaId Area ID to filter media players by
+ * @returns Array of active media player entities (playing or paused)
```
```suggestion
 * @returns Array of playing media player entities
```

winterborn committed:
Fix double computation in area card badge rendering

winterborn commented on src/data/media-player.ts:
```diff
@@ -520,3 +522,33 @@ export const mediaPlayerJoin = (
+export const computeActiveAreaMediaStates = (
+  hass: HomeAssistant,
+  areaId: string
+): HassEntityBase[] => {
```
Changes [here](https://github.com/home-assistant/frontend/pull/26893/commits)

winterborn commented on src/translations/en.json:
```diff
@@ -103,7 +103,9 @@
-        "area_not_found": "Area not found."
+        "area_not_found": "Area not found.",
+        "media_playing": "Media playing",
+        "media_paused": "Media paused"
```
Changes [here](https://github.com/home-assistant/frontend/pull/26893/commits)

piitaya commented:
Sorry, we had to revert the PR. After discussion with the rest of the team, it should implemented that way (more info in the 27413 PR)
